presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -243,13 +247,11 @@ public PlanNodeId getSourceId()
     public Supplier<Optional<UpdatablePageSource>> addSplit(ScheduledSplit split)
     {
         requireNonNull(split, "split is null");
-        checkState(this.taskSource == null, "NativeEngine operator split already set");
 
         if (finished) {
             return Optional::empty;
         }
-
-        this.taskSource = new TaskSource(split.getPlanNodeId(), ImmutableSet.of(split), true);
+        splitList.add(split);
 
         Object splitInfo = split.getSplit().getInfo();
```

Comment:
Is this logic correct? Shouldn't the setInfoSupplier be based on all splits, not the last one?

Reply:
This function `addSplit` is passed 1 split at a time, so `operatorContext.setInfoSupplier` will be called once for each split anyway.

Reply:
This is used in com.facebook.presto.operator.OperatorContext#getOperatorStats and will produce incomplete stats, no?

Reply:
As discussed I have added a `TODO` for this section. #19279

Reply:
I have added this in line 259